src/app/query/bitcoin/ordinals/use-taproot-address-utxos.query.ts

```diff
@@ -0,0 +1,48 @@
+import { useQuery } from '@tanstack/react-query';
+
+import { useCurrentTaprootAccountKeychain } from '@app/store/accounts/blockchain/bitcoin/taproot-account.hooks';
+import { useBitcoinClient } from '@app/store/common/api-clients.hooks';
+import { useCurrentNetwork } from '@app/store/networks/networks.selectors';
+
+import { UtxoResponseItem } from '../bitcoin-client';
+import { getTaprootAddress, hasOrdinals } from './utils';
+
+const stopSearchAfterNumberAddressesWithoutOrdinals = 5;
+
+// TODO: check phrasing of below comment
+/**
+ * Returns all utxos for the address associated with the user's current account.
+ * The search for utxos iterates through all addresses until a sufficiently
+ * large number of empty addresses is found.
+ */
+export function useTaprootAddressUtxosQuery() {
+  const network = useCurrentNetwork();
+  const keychain = useCurrentTaprootAccountKeychain();
+  const client = useBitcoinClient();
+
+  return useQuery(['transaction-metadata', keychain.pubKeyHash, network.id], async () => {
```

Comment:
If I saw this in the query devtools I wouldn't immediately know this was for TR utxos. Maybe a name that follows the hook would be better?